### PR TITLE
Emily/bottom play bar

### PR DESCRIPTION
**Pull request by emilysunaryo**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### :cloud: :cloud: Description :cloud: :cloud:
[//]: # "REQUIRED - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
This PR includes a new BottomPlayBar modal that will eventually be able to play and pause the current audio on every screen. In order to implement this, we had to rearrange our Navigation stack and have "PlayScreen" live outside of the "Navigation Bar" screen in RootNavigator.tsx

### Screenshots :calling:
[//]: # "REQUIRED for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! If you are making changes here, please CC: @leexesther at the bottom."
Kapture 2023-04-19 at 20 41 37

## :zap: How to review :zap:
[//]: # 'REQUIRED - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
Navigation.tsx could be potentially reviewed. We wrapped each navigation stack in CompositeScreenProps to fix the nagivation props error in each screen.

## :seedling: Next steps :seedling:
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
Next steps will be to implement audio playing functionality within this modal. We need to be able to pass the current audio object to this modal for it to use audio util controls like pause and play through every screen. The modal could also use some extra styling to center its components

### Online sources :information_source:
[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs :raised_hands:
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

[//]: # 'This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
CC: @davidqing6432

### CODE REVIEW

File: src/components/BottomPlayBar.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
+import { Image, Pressable, StyleSheet, Text, View } from 'react-native';
+import Icon from '../../assets/icons';
+
+const styles = StyleSheet.create({
+  card: {
+    backgroundColor: '#D9D9D9',
```
Comment (wangannie) [inline src/components/BottomPlayBar.tsx:6]: If there's a corresponding color in `Colors.ts`, use that! If it doesn't exist but there should be one, feel free to define a new color constant there.

File: src/i18n/i18n.ts [hunk 2]
```diff
@@ -7,6 +7,7 @@ const createI18n = (language: string): i18nInstance => {
   const i18n = i18next.createInstance().use(initReactI18next);
 
   i18n.init({
+    compatibilityJSON: 'v3',
```
Comment (wangannie) [inline src/i18n/i18n.ts:10]: This change shouldn't be relevant here

File: src/screens/PlayScreen/Play.tsx [hunk 3]
```diff
@@ -38,6 +72,18 @@ function PlayScreen() {
       </Text>
       <Text style={styles.author_text}>Shaldon Ferris</Text>
 
+      {/* <RectButton
```
Comment (wangannie) [inline src/screens/PlayScreen/Play.tsx]: Remove?

File: src/types/navigation.tsx [hunk 4]
```diff
@@ -25,17 +22,25 @@ export type GrantsStackParamList = {
...
   Audio: undefined;
-  Play: undefined;
 };
 
 export type RootStackScreenProps<T extends keyof RootStackParamList> =
   NativeStackScreenProps<RootStackParamList, T>;
 
 export type HomeStackScreenProps<T extends keyof HomeStackParamList> =
-  NativeStackScreenProps<HomeStackParamList, T>;
+  CompositeScreenProps<
```
Comment (wangannie) [inline src/types/navigation.tsx:31]: Would recommend adding a comment here to explain why we do this -- something like 

`// Note: we use composite screen props to support navigating to the 'Play' screen from any stack`

File: src/screens/PlayScreen/Play.tsx [hunk 5]
```diff
@@ -1,9 +1,41 @@
...
+import { Audio } from 'expo-av';
+import React from 'react';
 import Icon from '../../../assets/icons';
 import Colors from '../../styles/Colors';
 import styles from './styles';
+import { RootStackScreenProps } from '../../types/navigation';
+
+// TODO: Use toggleAudio when it is necessary, may be fixed in later PRs.
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+async function toggleAudio(
```
Comment (wangannie) [inline src/screens/PlayScreen/Play.tsx]: Remove this function for now -- you can find it in the commit history in this PR if you need it later.

File: src/components/BottomPlayBar.tsx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+function BottomPlayBar({ onPress, children, style }: BottomPlayBarProps) {
+  return (
+    <Pressable style={[styles.card, style]} onPress={onPress}>
+      {children}
+    </Pressable>
+  );
+}
+
+type SearchBottomPlayBarProps = {
```
Comment (wangannie) [inline src/components/BottomPlayBar.tsx:39]: I would rename this to `NowPlayingBar` or something for consistency with your wrapper naming. Also bc this bar doesn't actually handle any of the positioning logic, so including the word 'bottom' could be misleading.

**Review by wangannie — COMMENTED** [review]
Looks good! Ping me to review again once you finalize the styling of the play bar and add the wrapper logic.

Also, I would rename the PR to word it like a commit message -- look at other PR names for examples!
